presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/DynamicMetastoreConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.thrift;
+
+import com.facebook.airlift.configuration.Config;
+import com.facebook.airlift.configuration.ConfigDescription;
+
+public class DynamicMetastoreConfig
+{
+    private String metastoreUsername;
+    private ThriftMetastoreHttpRequestDetails metastoreDiscoveryUri;
+    private String metastoreDiscoveryType;
```

Comment:
We usually use enums

Reply:
The value of metastoreDiscoveryType is defined in hive.properties and set in DynamicMetastoreConfig when the metastore module is bootstrapped. Can you help me understand what can be set as an enum here? metastoreDiscoveryType is only used in the getter and setter.

Reply:
Create a new Enum called `MetastoreDiscoveryType` with two options: `STATIC` and `DYNAMIC`. We don't use strings to enums in configs. Then here, we will have `private MetastoreDiscoveryType metastoreDiscoveryType;`